semantic-python/src/Language/Python/Core.hs

```diff
@@ -201,10 +202,32 @@ deriving instance Compile Py.CompoundStatement
 instance Compile Py.ConcatenatedString
 instance Compile Py.ConditionalExpression
 instance Compile Py.ContinueStatement
-instance Compile Py.DecoratedDefinition
+
+instance Compile Py.DecoratedDefinition where
+  compile it@Py.DecoratedDefinition
+    { definition
+    , extraChildren = [ Py.Decorator { extraChildren } ]
+    } cc next = do
+    let thenReassign item = do
+          _ :> lastbound <- asks unBindings
+          tocall <- compile extraChildren pure next
+          let callit go = (pure lastbound .= (tocall $$ pure lastbound)) >>> go
+          fmap callit (cc item)
```

Comment:
What do you think about moving this into a `where` clause instead? `let`s with `do` blocks _within_ `do` blocks feel a little harder to read to me, but it’s pretty subjective.